src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ReferenceDirectiveCompletionProviderTests.cs

```diff
@@ -33,12 +35,12 @@ protected override bool CompareItems(string actualItem, string expectedItem)
         protected override Task VerifyWorkerAsync(
             string code, int position, string expectedItemOrNull, string expectedDescriptionOrNull,
             SourceCodeKind sourceCodeKind, bool usePreviousCharAsTrigger, bool checkForAbsence,
-            int? glyph, int? matchPriority, bool? hasSuggestionItem)
+            int? glyph, int? matchPriority, bool? hasSuggestionItem, IDictionary<OptionKey, object> options)
```

Comment:
As I said, I'm not too excited about options being involved in completion. What is the reason for that?

Reply:
> Could you explain why you think it's necessary?

I told VS to generate code according to a particular style. I expect my preferences to be honoured, and they are, with the exception of code generated by the completion providers. To fix that, the providers need to know what the current set of preferences are.

Reply:
I don't see a reason to do any complciated code generation here based on your preferred style.
We offer the name even if it doesn't exist:
![image](https://user-images.githubusercontent.com/11444821/37253756-22bd9446-2536-11e8-8f87-81df33d25cea.png)

Reply:
Is there a reason why this couldn't be entirely syntax based?

Reply:
You're focusing on one very specific, simple case. What would you expect to see for an argument to a method call?

```csharp
Foo(new $$)
```

Or an assignment to a property?

```csharp
x.Foo = new $$
```

How would you expect this to behave when the methods, properties and types involved are defined in a referenced assembly (so you don't know what the original code style is), or what happens if they're defined in a project, but that project has an editorconfig style that conflicts with the editorconfig used for the document you're typing in. Also consider that the type your instantiating may not be in any of your imported namespaces, so an extra resolution and minimizing process is applied.

I'm not saying it's impossible to do what you want, but there are a lot of cases where an obvious raw textual syntax is not readily available. And that's just considering the object-creation provider. In those cases you have to pick a code style, and that style should match what the user has set in their preferences.

Now, if you want to build on that for the special cases where a specific syntax _is_ available, then go ahead, I certainly wouldn't object.

Reply:
@lorcanmooney Alright thanks for the explanation, it makes more sense now. I thought this completion was specific to declaring a variable. I agree it's a lot more complicated if we're offering this everywhere. Sorry for bothering you